### PR TITLE
fix: [PIE-4585]: Custom artifact source

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts [hunk 1]
```diff
@@ -119,7 +119,8 @@ export const allowedArtifactTypes: Record<ServiceDefinition['type'], Array<Artif
     ENABLED_ARTIFACT_TYPES.Gcr,
     ENABLED_ARTIFACT_TYPES.Ecr,
     ENABLED_ARTIFACT_TYPES.Nexus3Registry,
-    ENABLED_ARTIFACT_TYPES.ArtifactoryRegistry
+    ENABLED_ARTIFACT_TYPES.ArtifactoryRegistry,
+    ENABLED_ARTIFACT_TYPES.CustomArtifact
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts:123]: This is already added, and is behind FF

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/CustomArtifact/CustomArtifact.tsx [hunk 2]
```diff
@@ -15,43 +15,247 @@ import {
...
-  initialValues,
   previousStep,
-  artifactIdentifiers,
   isReadonly = false,
-  selectedArtifact
-}: StepProps<ConnectorConfigDTO> & ImagePathProps<ImagePathTypes>): React.ReactElement {
+  formik
+}: any): React.ReactElement {
   const { getString } = useStrings()
+  const scriptType: ScriptType = formik.values?.spec?.shell || 'Bash'
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/CustomArtifact/CustomArtifact.tsx:71]: Use string const

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/CustomArtifact/CustomArtifact.tsx [hunk 3]
```diff
@@ -15,43 +15,247 @@ import {
...
+            />
+          </MultiTypeFieldSelector>
+          {getMultiTypeFromValue(formik.values?.spec?.scripts?.fetchAllArtifacts?.spec?.source?.spec?.script) ===
+            MultiTypeInputType.RUNTIME && (
+            <ConfigureOptions
+              value={formik?.values?.spec.source?.spec?.script as string}
+              type="String"
+              style={{ marginTop: 27 }}
+              variableName="spec.scripts.fetchAllArtifacts.spec.source.spec.script"
+              // className={css.minConfigBtn}
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/CustomArtifact/CustomArtifact.tsx:143]: remove commented code

**Review by bhavya-sinha — CHANGES_REQUESTED** [review]

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10174)

**Comment by satyam-harness** [comment]
trigger cypress

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 4]
```diff
@@ -87,8 +87,38 @@ export interface ImagePathTypes {
   repositoryFormat?: string
 }
 
-export interface CustomArtifactSource extends ImagePathTypes {
-  version: string
+export interface variableInterface {
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:90]:
```suggestion
export interface VariableInterface {
```
Interface must start with uppercase

**Review by bhavya-sinha — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**